### ISSUE TITLE
Make new release?

### ISSUE THREAD

**Issue by v1993** [pavouk/lgi#225]
It's been a while since last release (about 1.5 years) and lgi got some improvements meanwhile.

Can you, please, consider making new release and uploading it to luarocks? Thank you!

**Comment by v1993** [comment]
Bumping this. This is even more important than before considering further fixes and improvements, for me mostly but not limited to Lua 5.3 integers support.

**Comment by v1993** [comment]
Another bump... It's been 3 years already. @pavouk?

**Comment by v1993** [comment]
Okay, by this point we probably should make another lgi luarocks package if old one is no longer maintained or contact Luarocks staff to take over old one. It have been WAY too long without release.

**Comment by psychon** [comment]
@v1993 I have no prior experience with luarocks, but I'd suggest to make a fork. No idea what a good name would be, but
- I only have some access rights on this repo and @pavouk seems AWOL
- I no longer use LGI myself

**Comment by ousia** [comment]
I don’t know whether I may be missing something (or even the whole thing 😅).

After #280 is merged (and since we may have GTK4 support [not sure about #278]), could we have a new release?

Many thanks for your help and your excellent work.

**Comment by psychon** [comment]
> could we have a new release?

Sure, feel free to do one. I guess you are in as good a situation to do a release as I am.

@Elv13 @Aire-One AFAIK you should have as much permissions here as I do. Any takers?

**Comment by ousia** [comment]
> > could we have a new release?
>
> Sure, feel free to do one. I guess you are in as good a situation to do a release as I am.

@psychon,

many thanks for your generosity and the invitation.

I should have no permissions to make a new release (after all, I’m not even a collaborator [only a contributor]).

Besides, I have no proper programming knowledge (very basic _Lua_ scripts with only very basic variables and functions).

All I know from that project is that it contains the _Lua_ bindings for GTK+. It would be simply crazy that I released a new version. 

Many thanks for your help.

**Comment by Aire-One** [comment]
> [...] @Aire-One AFAIK you should have as much permissions here as I do. Any takers?

I'm only member at the LGI organization. (I think even my PR's approvals doesn't weight that much (at least, they haven't this green background like yours 🤔))

I guess I can still update the README changelog section and push a new tag 🤷

**Comment by Aire-One** [comment]
> I'm only member at the LGI organization. (I think even my PR's approvals doesn't weight that much (at least, they haven't this green background like yours thinking))
>
> I guess I can still update the README changelog section and push a new tag shrug

@psychon @pavouk, I can confirm I don't have the sufficient privilege elevation to push a new tag.

I have updated the README History in https://github.com/Aire-One/lgi/tree/prepare-0.10.0 based on the result of `git log 0.9.2..master --reverse --merges --grep='Merge pull request #'` ant cherry-pick of the “useful” commits.

> it seems that now you are the only member in LGI devs (https://github.com/orgs/lgi-devs/people).

Nahh, don't worry... I'm the only _public_ member, but there are other.

**Comment by Thomashighbaugh** [comment]
@ousia 

Obviously I am not a contributor, nor have any membership in the LGI organization here or GitHub or otherwise but I thought it might be worthwhile to point out that while there has not been a new release in a while there is no reason to consider the project abandoned or even stalled, just the victim of people going AWOL without assigning to others a sufficient amount of permissions to actually create new releases (alternatively they could have assigned such permissions to no one else and forced the development to continue in the form of forks, which in some cases works well enough but in most cases leads to the project dying like "Way Cooler"). 

Some projects that are still used, such as the `runit` init system used by Void Linux and Artix Linux (among others), are dormant in terms of the underlying code as there is simply no reason for any changes to be made in its deliberately minimal code base. That's not a sign of project death but a pinnacle that some projects aim to reach, one where the code is at a place that it can continue to serve its intended functions into perpetuity without any more work from the developers. A sort of open-source development nirvana if you will. 

But don't doubt yourself, if it is important to you to have it available via Luarocks, you are probably in the best place to maintain the package via releases and can pull any additional developments in this repo to your fork easily enough. You don't need an advanced degree in computer science to do that sort of thing, honestly with how scatter brained the people with those degrees are you might actually be in a better place without the thing. 

What makes you a true `coder` or `dev` in the open source community is when no one else is doing something and you need the thing enough, through sheer force of irritation you just do it yourself. There is no government licenses issued for software developers (that I know of at least), so you are one if you consider yourself as much. Like the old Greeks said, "how many lies does it take before one becomes a liar? If he tells one less lie than that, is he still an honest man?" Reformulating that for our context here: `how many lines of code does it take for one to be a coder? If he writes one less line than that is he still technically illiterate?` This of course drives at the point, you are already sufficiently skilled to consider yourself a developer and quite capable of publishing releases from your own fork if you even know what Lua is, I believe in you. 

As for a name for a potential fork, my advise would be to use `lgi-ng` for next generation, or something like that. 

### Hope This Finds You All Well
## Thanks for the Hard Work on the Project to All Involved

**Comment by ziyao233** [comment]
It has been 2025 now, 8 years from the latest release. Lua 5.4 introduces API breakage, thus lgi 0.9.2 doesn't build with. A stable release also benefits distro packaging.

**Comment by actionless** [comment]
bumping the topic as it's getting ridiculous (same ridiculous btw as https://github.com/awesomeWM/awesome/issues/3937 ) and instead of being stable both simply being outdated

**Comment by mrcjkb** [comment]
I don't have anything to do with this project, but I do have some experience with luarocks packages that have lost their original maintainers. 

The issue is that you don't only need permissions to push a new tag, you'll also need permissions to push to [@pavouk's luarocks account](https://luarocks.org/modules/pavouk).

So here's what I'd recommend if you want to maintain releases:

- Fork this project and request GitHub to detach it. 
- In the luarocks-site repository, open an issue requesting to take over the `lgi` package. Here are some examples: https://github.com/luarocks/luarocks-site/issues?q=is%3Aissue%20state%3Aopen%20label%3A%22module%20issue%22
This can take a while because you'll need to verify that the original maintainers are in fact no longer reachable (e.g. by trying to reach them outside of GitHub).

An alternative would be to fork and publish under a different name.

You can use something like the [luarocks-tag-release GitHub action](https://github.com/lumen-oss/luarocks-tag-release/) to automate publishing to luarocks.

**Comment by shamefulCake1** [comment]
I has been 9 years.

**Comment by heirecka** [comment]
I have no particular experience with lgi (other than that I use it), but I do releases for >= 200 packages included with KDE Gear. So I hopefully know a bit about releasing software and I'm happy to help with getting a release done if there's a need for this help.

**Comment by Aire-One** [comment]
I think the most up-to-date topic on this matter is https://github.com/awesomeWM/awesome/issues/3338#issuecomment-5318069844.

Quoting :

> Creating a new tag is possible for me since very recently with https://github.com/lgi-devs/lgi/pull/361#issuecomment-5093211436 [...]
> 
> Soooo, technically, yeah I should be able to work on some newer version release and tag it on the LGI repo. Will try my hands next time I'm bored [...]

I did start to work on that a while ago (like multiple years ago) with https://github.com/Aire-One/lgi/tree/prepare-0.10.0 where I have listed some notable changes since last release. To be completed with new stuff done since then...

I guess we should also make sure the build toolchain use current versions of the tools before we tag a new release to make it easy for downstream to pick the tag.

**Comment by actionless** [comment]
@Aire-One and what regarding access to luarocks account which was presented previously as one of the release blockers, does that aspect also got resolved by now?

**Comment by Aire-One** [comment]
> [@Aire-One](https://github.com/Aire-One) and what regarding access to luarocks account which was presented previously as one of the release blockers, does that aspect also got resolved by now?

Good point. In https://github.com/lgi-devs/lgi/issues/302#issuecomment-1322746532, @leafo offered to change the org in Luarocks to help us be owner of the package. This is something we need to work with to allow us to push new rocks.

@leafo, what are the available options to us? Can the Luarocks repository be managed by multiple people from the GitHub lgi-dev organization members? Ideally, I would like to generate an organization level token to manage that automatically from the CI using https://github.com/lunarmodules/.github/blob/main/.github/workflows/upload_to_luarocks.yml
